src/game/PetAI.cpp

```diff
@@ -86,6 +86,18 @@ void PetAI::UpdateAI(const uint32 diff)
     else
         m_updateAlliesTimer -= diff;
 
+    // Prevent losing the pet action bar after reloading the user interface
```

Comment:
You can use this OpCodeHandler:
void WorldSession::HandleRequestPetInfoOpcode( WorldPacket & /*recv_data */)
{
    /*
        DEBUG_LOG("WORLD: CMSG_REQUEST_PET_INFO");
        recv_data.hexlike();
    */
}
And simply put in _player->PetSpellInitialize();